### PR TITLE
Refactor installation procedure

### PR DESCRIPTION
**Pull request by gsanchietti**
- Remove extra partition: reduce the image size by a half, installation now requires Internet access
- Remove install option from menu: the procedure will be documented
- Remove 'Exit' option from menu since it's not needed
- Install script renamed from `install-nx.sh` to `ns-install`
- Install script is now interactive (code partially imported from menu script)
- Install script can be executed in batch mode: `ns-install -t /dev/sdb -f`
- Add `ns-reset` script to execute factory default, the script requires Internet access to download the image
- Add `ns-download` script to download current or latest image version 
- Remove login menu

### CODE REVIEW

File: files/usr/bin/ns-download [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    case $opt in
+        (f) force=1 ;; # force download
+        (l) latest=1 ;; # download latest image
+        (*) help ;;
+    esac
+done
+
+# Force download by deleting destination dir
+if [ "$force" -eq 1 ]; then
+    rm -rf "$DL_DIR" &> /dev/null
```
Comment (filippocarletti) [inline files/usr/bin/ns-download:43]:
```suggestion
    rm -rf "$DL_DIR"
```

File: files/usr/bin/ns-download [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+img_name="nextsecurity-$version-$ARCH-generic-ext4-combined-efi.img.gz"
+img="$DL_DIR/$img_name"
+img_url="$BASE_URL/$version/targets/$OPENWRT_BOARD/$img_name"
+hash_url="$BASE_URL/$version/targets/$OPENWRT_BOARD/sha256sums"
+
+# Download if image does not exists
+if [ ! -d "$DL_DIR" ] || [ ! -f "$img" ] || [ ! -f "$hash" ]; then
+    mkdir -p "$DL_DIR"
+    rm -f "$img" "$hash" &> /dev/null
+    curl -L -s -m 60 -o "$hash" "$hash_url"
```
Comment (filippocarletti) [inline files/usr/bin/ns-download:66]: 60 seconds to download an hash seems like an eternity :-)

File: files/usr/bin/ns-download [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    curl -L -s --connect-timeout 60 -o "$img" "$img_url"
```
Comment (filippocarletti) [inline files/usr/bin/ns-download:67]: Also 60 seconds for the connect is a really big time

File: files/usr/bin/ns-download (hunk 2 above)
Comment (gsanchietti) [reply files/usr/bin/ns-download:66]: What about 30?

File: files/usr/bin/ns-download (hunk 3 above)
Comment (gsanchietti) [reply files/usr/bin/ns-download:67]: What about 20?

File: files/usr/bin/ns-download (hunk 2 above)
Comment (gsanchietti) [reply files/usr/bin/ns-download:66]: fixed

File: files/usr/bin/ns-download (hunk 1 above)
Comment (gsanchietti) [reply files/usr/bin/ns-download:43]: fixed

**Pull request merged**